server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregator.java

```diff
@@ -218,6 +222,10 @@ protected Function<Long, Long> bucketOrdProducer() {
                 return (key) -> getBucketOrds().add(0, preparedRounding.round(key));
             }
         };
+
+        this.fieldName = (valuesSource instanceof ValuesSource.Numeric.FieldData)
+            ? ((ValuesSource.Numeric.FieldData) valuesSource).getIndexFieldName()
+            : null;
```

Comment:
Maybe better to not change anything in `AutoDateHistogramAggregator`? You can keep the diff as patch for reference going forward